### PR TITLE
Supporting passing Kafka basic auth creds in environment variables

### PR DESCRIPTION
**Pull request by delthas**
Used by Zenko, that uses environement variables rather than a configuration file currently.

Issue: BB-732

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -39,6 +39,7 @@ FROM node:${NODE_VERSION}
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
+        ca-certificates \
```
[the remark this answers is not in the document]
Comment (SylvainSenechal) [reply Dockerfile:42]: Nit but agree with the bot : could be a separate commit with a message to explain why its added

**Review by SylvainSenechal — APPROVED** [review]
Looks good, for some reason I really struggle to read Joi validation code

File: extensions/notification/utils/auth.js [hunk 2]
```diff
@@ -85,19 +85,26 @@ function generateKafkaAuthObject(auth) {
...
+                    throw new Error(`Credentials file ${credentialsFile} not found in ${authFilesFolder}`);
+                }
+                const credentials = readCredentialsFile(credsFilePath);
+                authObject['sasl.username'] = credentials.username;
+                authObject['sasl.password'] = credentials.password;
+            } else if (username && password) {
+                authObject['sasl.username'] = username;
+                authObject['sasl.password'] = password;
+            } else {
+                throw new Error('Missing credentials');
```
Comment (francoisferrand) [inline extensions/notification/utils/auth.js]: can this really happen?
- if `credentialsFile` is set, we call `readCredentialsFile()` which validates the field are set indeed
- otherwise `username` & `password` should have been validated when validating the config with joi


```suggestion
			} else {
                authObject['sasl.username'] = username;
                authObject['sasl.password'] = password;
            }
```

File: tests/unit/notification/utils/auth.js [hunk 3]
```diff
@@ -144,6 +160,14 @@ describe('generateKafkaAuthObject', () => {
...
             },
         },
+        {
+            description: 'Basic authentication with missing credentials',
+            valid: false,
+            input: {
+                type: 'basic',
+                protocol: 'SASL_PLAINTEXT',
+            },
+        },
```
Comment (francoisferrand) [inline tests/unit/notification/utils/auth.js]: for completeness, should also have test case when only username of password is missing or empty, when `credentialsFile` is empty, and when both username/password and credentialsFile are specified

**Review by francoisferrand — APPROVED** [review]

File: tests/unit/notification/utils/auth.js (hunk 3 above)
Comment (delthas) [reply tests/unit/notification/utils/auth.js]: Added

File: extensions/notification/utils/auth.js (hunk 2 above)
Comment (delthas) [reply extensions/notification/utils/auth.js]: Hm indeed. I got confused because I wrongly created my unit tests against generateKafkaAuthObject, meaning after the validation. I moved the unit tests to target the validation and now they properly pass without this added else case.
